### PR TITLE
Add explanation of and plans for repo to kb_sdk_plus.md

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: kb_sdk_plus.md [hunk 1]
```diff
@@ -1,10 +1,98 @@
...
+    * It might be possible to make backwards incompatible changes that are fixed
+      with a recompile of an app or a simple migration guide.
+      This would need careful thought and implementation.
+* Generally simplify the codebase where possible and document where needed
+      
+## Prior to releasing for general development
+
+* Get the tests to pass locally
+  * Remove R support, probably remove Perl support
+  * Do we still need Javascript support?
```
Comment (MrCreosote) [inline kb_sdk_plus.md:41]: @briehl @ialarmedalien wdyk?

File: kb_sdk_plus.md [hunk 2]
```diff
@@ -1,10 +1,98 @@
...
+
+`<repo>2` generally signifies a greenfield rewrite of a codebase in KBase,
+which `kb_sdk_plus` is not. It's forking a codebase to leave the original codebase untouched
+and runnable while we make changes to the forked codebase.
+
+## Guidelines
+
+* Backwards incompatible changes are expected in some respects:
+  * R code generation will be removed
+    * Unused in KBase or anywhere else as far as we know
```
Comment (ialarmedalien) [inline kb_sdk_plus.md:24]: There are at least two R KBase apps, but at least one, possibly both, are called from the command line so the fact that they are in R is irrelevant.

File: kb_sdk_plus.md [hunk 3]
```diff
@@ -1,10 +1,98 @@
...
+* Get `kb_sdk_plus` image build running in GHA
+  * Or we could just call it `kb_sdk` since the original container is on Dockerhub
+* Add tests that run `kb-sdk` from the command line vs. just testing java classes
+  * Many issues crop up here
+* Get the SDK running on docker03
+  * Not sure what the issues are here, need to investigate
+* Figure out how to support Argonne staff, where Docker Desktop is banned
+  * Is the docker engine CLI not enough?
+
+## Post general development release
```
Comment (ialarmedalien) [inline kb_sdk_plus.md:73]: Could also mention the issues from the kb_sdk repo here -- e.g. an annoying python bug concerning the location of the auth and base modules.

Comment (ialarmedalien) [reply kb_sdk_plus.md:73]: (Investigate if it is possible to move/copy issues to this repo)

File: kb_sdk_plus.md (hunk 2 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:24]: the only R app at https://narrative.kbase.us/legacy/catalog/modules is Roman's ContigFilter test app and it's in dev, so I think we can safely remove R support.

File: kb_sdk_plus.md (hunk 3 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:73]: I linked to the kb_sdk issues page; were you thinking of something more than that?

File: kb_sdk_plus.md (hunk 2 above)
Comment (MrCreosote) [reply kb_sdk_plus.md]: Added a note that one test app does exist for accuracy

**Review by ialarmedalien — APPROVED** [review]

File: kb_sdk_plus.md (hunk 3 above)
Comment (ialarmedalien) [reply kb_sdk_plus.md:73]: well, possibly something more explicit like "assess and migrate issues from <kb_sdk repo url>" but the link is fine

File: LICENSE.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2014 The KBase Project and its Contributors
+Copyright (c) 2014-present The KBase Project and its Contributors
```
Comment (briehl) [inline LICENSE.md:1]: Ugh, I hate to ask, but is there an official doc on this? Do we update to the current year, or is it enough to put `-present`?

File: kb_sdk_plus.md [hunk 5]
```diff
@@ -1,10 +1,98 @@
...
+## Why not kb_sdk2?
+
+`<repo>2` generally signifies a greenfield rewrite of a codebase in KBase,
+which `kb_sdk_plus` is not. It's forking a codebase to leave the original codebase untouched
+and runnable while we make changes to the forked codebase.
+
+## Guidelines
+
+* Backwards incompatible changes are expected in some respects:
+  * R code generation will be removed
```
Comment (briehl) [inline kb_sdk_plus.md:23]: 👍

File: kb_sdk_plus.md [hunk 6]
```diff
@@ -1,10 +1,98 @@
...
+    * Unused in KBase or anywhere else as far as we know other than one test app
+  * Perl code generation will almost certainly be removed
+    * Perl apps can continue to be supported with the original SDK, or wrapped with
+      Python / Java
+      * https://github.com/boriel/perlfunc for instance
```
Comment (briehl) [inline kb_sdk_plus.md:28]: Any examples of this in KBase? I don't think there are, just checking if anyone's tried that.

File: kb_sdk_plus.md (hunk 1 above)
Comment (briehl) [reply kb_sdk_plus.md:41]: What does "support" mean here? Just making clients? We have a bunch of different implementations of JS clients throughout the codebase, made for different purposes.

I'd like to normalize on one, but that feels like feature creep here, especially as the current SDK JS client code consists of an AJAX call + a name string and very little type checking or error handling.

I think I just convinced myself that we don't need to support it in this particular effort.

Comment (briehl) [reply kb_sdk_plus.md:41]: Also, once things are spruced up, it probably won't be a heavy lift to tack on JS client support, if that's desired.

File: kb_sdk_plus.md [hunk 7]
```diff
@@ -1,10 +1,98 @@
...
+* Ensure tests are running against EE2 vs. EE1 (e.g. the `KBaseJobService.spec`
+  file is suspicious)
+* Fix docker version issues (see Dakota's patch: https://github.com/kbase/kb_sdk_patch)
+* Convert the build to Gradle
+  * Remove `build.xml`, `travis.yml` and `.classpath`
+* Update to Java 11 at least. Test with newer Java versions if possible
+* Get the tests running in GHA with coverage, dependabot, and trivy
+  * At least look into running macOS and windows tests in GHA
+    * If not, clearly document manual testing requirements before a release
+    * Do we need to support windows?
```
Comment (briehl) [inline kb_sdk_plus.md:58]: That might be a question for Ben.

File: kb_sdk_plus.md [hunk 8]
```diff
@@ -1,10 +1,98 @@
...
+      * https://github.com/orgs/community/discussions/69211#discussioncomment-7197681
+      * https://github.com/actions/runner/issues/904
+      
+* Get `kb_sdk_plus` image build running in GHA
+  * Or we could just call it `kb_sdk` since the original container is on Dockerhub
+* Add tests that run `kb-sdk` from the command line vs. just testing java classes
+  * Many issues crop up here
+* Get the SDK running on docker03
+  * Not sure what the issues are here, need to investigate
+* Figure out how to support Argonne staff, where Docker Desktop is banned
```
Comment (briehl) [inline kb_sdk_plus.md:70]: AFAIK, Rancher Desktop is a decent drop-in solution and is, I think, what some ANL folk use.

File: LICENSE.md (hunk 4 above)
Comment (MrCreosote) [reply LICENSE.md:1]: With a quick look I can't find anything either way. I doubt it particularly matters given the looseness of the license. If you're really exercised about it we should ask legal

File: kb_sdk_plus.md (hunk 6 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:28]: I'm unaware of any

File: kb_sdk_plus.md (hunk 7 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:58]: I was going to ask exec later down the line

File: kb_sdk_plus.md (hunk 1 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:41]: My understanding is that the JS clients at this point are effectively generic JSONRPC clients, so there's no real reason for the SDK to generate anything

File: kb_sdk_plus.md (hunk 8 above)
Comment (MrCreosote) [reply kb_sdk_plus.md:70]: I don't really understand the Mac situation. Can you not just install the command line client and use that? I don't use a GUI for docker at all

File: LICENSE.md (hunk 4 above)
Comment (briehl) [reply LICENSE.md:1]: I don't really care that much, TBH. I'll let someone else bother legal. :)

**Pull request merged**